domains/auth/Auth.API/Unit.Tests/Controllers/LoginControllerTests.cs

```diff
@@ -48,10 +51,45 @@ public async Task LoginAsync_ShouldReturnRedirectToAuthority_WhenInvoked()
         Assert.Equal(options.Value.AuthorityUri.Host, uri.Host);
 
         var query = HttpUtility.UrlDecode(uri.Query);
+        Assert.Contains($"state=", query);
         Assert.Contains($"client_id={options.Value.ClientId}", query);
         Assert.Contains($"redirect_uri={options.Value.AuthorityCallbackUri.AbsoluteUri}", query);
     }
 
+    [Fact]
+    public async Task LoginAsync_ShouldReturnRedirectToAuthorityWithConfiguredState_WhenInvokedWithConfigurations()
+    {
+        var options = TestOptions.Oidc(oidcOptions);
+
+        var document = DiscoveryDocument.Load(new List<KeyValuePair<string, string>>() { new("authorization_endpoint", $"http://{options.Value.AuthorityUri.Host}/connect") });
+
+        var cache = Mock.Of<IDiscoveryCache>();
+        _ = Mock.Get(cache).Setup(it => it.GetAsync()).ReturnsAsync(document);
+
+        var logger = Mock.Of<ILogger<LoginController>>();
+
+        var state = Guid.NewGuid().ToString();
+        var redirectionUri = Guid.NewGuid().ToString();
+
+        var result = await new LoginController().LoginAsync(cache, options, logger, state, redirectionUri);
+
+        Assert.NotNull(result);
+        Assert.IsType<RedirectResult>(result);
+
+        var redirectResult = (RedirectResult)result;
+        var uri = new Uri(redirectResult.Url);
+        var query = HttpUtility.UrlDecode(uri.Query);
+        var map = QueryHelpers.ParseNullableQuery(query);
+
+        Assert.NotNull(map);
+        Assert.True(map.ContainsKey("state"));
+
+        var decoded = OidcState.Decode(map["state"]);
+        Assert.NotNull(decoded);
+        Assert.Equal(state, decoded.State);
+        Assert.Equal(redirectionUri, decoded.RedirectionUri);
```

Comment:
The state in the query string should be equal to original state. OidcState is only for the "inside".